### PR TITLE
Convert to use editor-core-types

### PR DESCRIPTION
**Pull request by glennsl**
Zero-based indexing used everywhere here

### CODE REVIEW

File: test/ArrayParserTest.re [hunk 1]
```diff
@@ -251,12 +247,14 @@ describe("ArrayParser", ({describe, _}) => {
...
           ~range=
-            Types.Range.createi(
-              ~startLine=2,
-              ~startColumn=0,
-              ~endLine=2,
-              ~endColumn=1,
-              (),
+            Range.create(
+              ~start=
+                Location.create(~line=Index.(zero + 2), ~column=Index.zero),
```
Comment (bryphe) [inline test/ArrayParserTest.re]: Thank you for converting all of these, @glennsl !

File: test/SyntaxTest.re [hunk 2]
```diff
@@ -37,19 +36,14 @@ describe("Syntax", ({describe, _}) => {
...
-      let endColumn = errorRange.endPosition.column;
+      expect.int((errorRange.start.line :> int)).toBe(0);
+      expect.int((errorRange.stop.line :> int)).toBe(0);
 
-      expect.int(startLine).toBe(0);
-      expect.int(endLine).toBe(0);
-
-      expect.int(startColumn).toBe(2);
-      expect.int(endColumn).toBe(3);
+      expect.int((errorRange.start.column :> int)).toBe(2);
```
Comment (bryphe) [inline test/SyntaxTest.re]: Interesting, I hadn't seen the coercion operator before `:>` or the `type t = pri ...` before, used in `editor-core-types`

Comment (glennsl) [reply test/SyntaxTest.re]: I used it for `NormalizedTime` in Revery too. Outside of tests it's probably better to be explicit though.

**Comment by glennsl** [comment]
Unsurprisingly I don't have commit rights to this repo either... :man_shrugging:

**Comment by bryphe** [comment]
Sorry about that! Should be fixed now. Will be bring in anyway though so you don't have to wait. Thanks for all the work to convert to `editor-core-types`!

**Pull request merged**